<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>2. Tutorial (Basic Usage) &mdash; MeRP 0.0.1 documentation</title>
    
    <link rel="stylesheet" href="../_static/default.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '0.0.1',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="top" title="MeRP 0.0.1 documentation" href="../index.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li><a href="../index.html">MeRP 0.0.1 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="tutorial-basic-usage">
<h1>2. Tutorial (Basic Usage)<a class="headerlink" href="#tutorial-basic-usage" title="Permalink to this headline">¶</a></h1>
<p>The MeRP pipeline works in the python console. First, from the command line, install the package.</p>
<div class="highlight-python"><div class="highlight"><pre><span class="gp">&gt;&gt;&gt; </span><span class="n">sudo</span> <span class="n">python</span> <span class="n">setup</span><span class="o">.</span><span class="n">py</span> <span class="n">install</span>
</pre></div>
</div>
<p>Then enter your python console</p>
<div class="highlight-python"><div class="highlight"><pre><span class="gp">&gt;&gt;&gt; </span><span class="n">python</span>
</pre></div>
</div>
<p>To import and initialize MeRP after entering the Python console, run the commands:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">merp</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">m</span> <span class="o">=</span> <span class="n">merp</span><span class="o">.</span><span class="n">Merp</span><span class="p">()</span>
</pre></div>
</div>
<p>In this tutorial we will address the following question:</p>
<p>Does LDL cholestrol levels have a causal effect on the risk of Coronary Heart Disease?</p>
<div class="section" id="pulling-genetic-instruments-from-gwas-catalog-data">
<h2>2.1. Pulling Genetic Instruments from GWAS Catalog Data<a class="headerlink" href="#pulling-genetic-instruments-from-gwas-catalog-data" title="Permalink to this headline">¶</a></h2>
<p>Run this python command first in the package directory to pull latest NHGRI GWAS catalog and create a set of IV trait files in the directory /traitFiles along with an index file mapping traits to IDs in merp directory.</p>
<div class="highlight-python"><div class="highlight"><pre><span class="gp">&gt;&gt;&gt; </span><span class="n">m</span><span class="o">.</span><span class="n">pull</span><span class="p">()</span>
</pre></div>
</div>
<div class="section" id="optional-argument">
<h3>2.1.1. Optional argument:<a class="headerlink" href="#optional-argument" title="Permalink to this headline">¶</a></h3>
<p>Default settings presented.</p>
<blockquote>
<div><ol class="arabic">
<li><dl class="first docutils">
<dt>keyword = &#8220;&#8221;</dt>
<dd><ul class="first last simple">
<li>If keyword specified, only IVFs with trait/disease name containing keyword will be generated in /traitFiles.</li>
</ul>
</dd>
</dl>
</li>
</ol>
</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Frozen version of catalog (in /data) and trait IVFs (/traitFiles) from 8/10/14 included in package. Running pull function will overwrite these files.</p>
</div>
</div>
</div>
<div class="section" id="selecting-updating-ivs-for-traits-of-interest">
<h2>2.2. Selecting/Updating IVs for Traits of Interest<a class="headerlink" href="#selecting-updating-ivs-for-traits-of-interest" title="Permalink to this headline">¶</a></h2>
<p>Browse the &#8220;index&#8221; file in the home directory to find ID numbers associated with traits of interest.</p>
<p>Example Index</p>
<a class="reference internal image-reference" href="_sources/_static/index.png"><img alt="_sources/_static/index.png" class="align-center" src="_sources/_static/index.png" style="width: 400px;" /></a>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">MeRP pipeline is specifically designed for traits and biomarkers that are measured in natural log of odds ratio (beta) and not in odds ratio in order to run the analysis with disease endpoints. Please keep this in my mind when selecting trait of interests, and convert odds ratios to betas before analysis.</p>
</div>
<p>Run update function with trait file(s) of interest to add in non-risk allele and assign directional values to the beta effects according to the units description.</p>
<p>To update LDL cholesterol file, we run:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="gp">&gt;&gt;&gt; </span><span class="n">m</span><span class="o">.</span><span class="n">update</span><span class="p">(</span><span class="s">&#39;traitFiles/472&#39;</span><span class="p">)</span>
</pre></div>
</div>
<dl class="docutils">
<dt>Output:</dt>
<dd><ul class="first last simple">
<li>Updated file added to /traitFiles as 472_update.</li>
<li>List of SNPs without allele info from 1000 genomes if applicable. Non-risk allele no reported. [NR]</li>
</ul>
</dd>
</dl>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">While we use 1000 genomes SNP data to find non-risk alleles for SNPs, there are inevitably SNPs that are missing data. Due to this limitation, SNP lines with [NR] for non-risk allele should either be filled in with independent look up (dbSNP, etc.)or deleted before proceeding with filter().</p>
</div>
</div>
<div class="section" id="filter-trait-ivfs-for-confounding-and-ld">
<h2>2.3. Filter Trait IVFs for Confounding and LD<a class="headerlink" href="#filter-trait-ivfs-for-confounding-and-ld" title="Permalink to this headline">¶</a></h2>
<blockquote>
<div><div class="highlight-python"><div class="highlight"><pre><span class="gp">&gt;&gt;&gt; </span><span class="n">m</span><span class="o">.</span><span class="n">filter</span><span class="p">(</span><span class="s">&#39;traitFile 472_update’,nhgri_ignore=&#39;</span><span class="n">confounders</span><span class="o">/</span><span class="n">nhgri_ignore</span><span class="o">.</span><span class="n">txt</span><span class="s">&#39;,confounders=&#39;</span><span class="n">confounders</span><span class="o">/</span><span class="n">confounders</span><span class="o">.</span><span class="n">txt</span><span class="s">&#39;,primary_confounders=&#39;</span><span class="n">confounders</span><span class="o">/</span><span class="n">prim_confounders_chd</span><span class="o">.</span><span class="n">txt</span><span class="s">&#39;]</span>
</pre></div>
</div>
<dl class="docutils">
<dt>Output:</dt>
<dd><ul class="first last simple">
<li>Trait file filtered for confounding and correlation generated in : /traitFiles/461_updatefiltered</li>
</ul>
</dd>
</dl>
</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The filter will not work if there are still [NR] non-risk alleles or other format issues.
Users can input their own trait files in the filter as long as it is properly formatted. See Detailed Workflow for details on file formats.</p>
</div>
<div class="section" id="nhgri-ignore-and-confounders">
<h3>2.3.1. nhgri_ignore and confounders<a class="headerlink" href="#nhgri-ignore-and-confounders" title="Permalink to this headline">¶</a></h3>
<p>These two required text files should be kept in the merp directory, /confounders, and consist of a list of items separated by new lines.</p>
<p>nhgri_ignore.txt: list of traits that are found in NHGRI catalog that may be related to or similar to trait of interest but is not confounding.</p>
<div class="figure align-center">
<a class="reference internal image-reference" href="_sources/_static/nhgri_ignore.png"><img alt="_sources/_static/nhgri_ignore.png" src="_sources/_static/nhgri_ignore.png" style="width: 307px;" /></a>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">You may run filter function once with blank nhgri_similar.txt file. The program will print out exclusions because of associations in NHGRI catalog. Select those that are similar to trait of interest and that you DON&#8217;T want SNPs to be tossed out for.</p>
</div>
<p>confounders.txt: list of column headers from pval_file that are potentially related to trait of interest or endpoint disease. Default use is all metabolic pval file (see detailed workflow)</p>
<div class="figure align-center">
<img alt="_sources/_static/confounders.png" src="_sources/_static/confounders.png" />
</div>
<p>primary_confounders.txt (optional): list of column headers from pval_file that we would like to be more strict on. May be known to directly affect disease endpoint</p>
<div class="figure align-center">
<img alt="_sources/_static/prim_confounders.png" src="_sources/_static/prim_confounders.png" />
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Package includes list of all confounders in /confounders/confounders.txt of default pval file. Optional primary confounder file is formatted the same way.</p>
</div>
</div>
<div class="section" id="optional-arguments">
<h3>2.3.2. Optional arguments:<a class="headerlink" href="#optional-arguments" title="Permalink to this headline">¶</a></h3>
<p>Default settings presented.</p>
<blockquote>
<div><ol class="arabic">
<li><dl class="first docutils">
<dt>primary_confounders = &#8220;&#8221;</dt>
<dd><ul class="first last simple">
<li>String with path/to/file</li>
<li>If keyword specified, only IVFs with trait/disease name containing keyword will be generated in /traitFiles.</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>verbose = True</dt>
<dd><ul class="first last simple">
<li>If False, does not print detailed logs. Text logs always generated in /logs.</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>out = False</dt>
<dd><ul class="first last simple">
<li>If True, means user is bringing in outside IVFs not generated from NHGRI catalog.</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>localp = False</dt>
<dd><ul class="first last simple">
<li>Default setting streams metabolic pval_file from our server. To use local copy or customized pval_file at /data/pval_file.txt, set to True.</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>pmaxprimary = 0.01</dt>
<dd><ul class="first last simple">
<li>if a SNP has a single violation of p&lt;pmaxprimary, SNP excluded</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>pmax1 = 0.05</dt>
<dd><ul class="first last simple">
<li>if a SNP has more than threshold1 number of associations with p&lt;pmax1, SNP excluded.</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>threshold1 = 3</dt>
<dd><ul class="first last simple">
<li>if a SNP has more than threshold1 number of associations with p&lt;pmax1, SNP excluded.</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>pmax2 = 0.001</dt>
<dd><ul class="first last simple">
<li>if a SNP has more than threshold2 associations of pmax2, SNP excluded.</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>threshold2 = 0</dt>
<dd><ul class="first last simple">
<li>if a SNP has more than threshold2 associations of pmax2, SNP excluded.</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>rsq_threshold = 0.05</dt>
<dd><ul class="first last simple">
<li>SNPs with R^2 value &gt; rsq_threshold are clustered together, with only lead SNP passing on.</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>max_fraction = 0.1</dt>
<dd><ul class="first last simple">
<li>If number of total pmax1 violations exceeds max_fraction * total number of tests, then cut SNPS until below max_fraction of tests.</li>
</ul>
</dd>
</dl>
</li>
</ol>
</div></blockquote>
<p>As shown above, LDL Cholesterol, our nhgri_ignore.txt file contains traits found in NHGRI catalog related to LDL. Our confoudners.txt file contains headers that are potential confoudners and primary_confounders.txt contains headers that are known to associate with coronary heart disease.</p>
</div>
<div class="section" id="units">
<h3>2.3.3. Units<a class="headerlink" href="#units" title="Permalink to this headline">¶</a></h3>
<p>Due to the current nature of the NHGRI GWAS catalog, there may be several different units, or variations on the same unit present in the filtered IVF. At the end of the filter, we output the different units that must be addressed before moving on to IV calculations. For accurate estimations using calc() the units must all agree. Users may acheive this by tossing out SNPs or applying appropriate conversions and manually editing beta and unit values in a text editor.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The unit &#8220;units&#8221; most often means standard deviation units. As so s.d, sd, sd_%, etc. Be aware of these and use your best judgement when fixing units.</p>
</div>
<p>For the purposes of our tutorial, we delete SNP lines that are not in s.d units and those without non-risk allele in 1000 genomes data before proceeding to calculation.</p>
</div>
</div>
<div class="section" id="calculate-trait-disease-casual-effect-estimate">
<h2>2.4. Calculate Trait-Disease Casual Effect Estimate<a class="headerlink" href="#calculate-trait-disease-casual-effect-estimate" title="Permalink to this headline">¶</a></h2>
<p>Lastly, we calculate the estimated causal effect of LDL cholesterol on Coronary Heart Disease (CHD) using our filtered trait IVF along with the provided CVD gwas summary file.</p>
<div class="highlight-python"><div class="highlight"><pre><span class="gp">&gt;&gt;&gt; </span><span class="n">m</span><span class="o">.</span><span class="n">calc</span><span class="p">(</span><span class="s">&#39;traitFile/472_update&#39;</span><span class="p">,</span> <span class="s">&#39;data/demo_chd.txt&#39;</span><span class="p">)</span>
</pre></div>
</div>
<dl class="docutils">
<dt>Output:</dt>
<dd><ul class="first last simple">
<li>Summary and individual SNP result file with estimated causal effect (alpha), standard error on effect, and p-value on result.</li>
</ul>
</dd>
</dl>
<p>The summary and individual result files can be found at /analysis</p>
<p>472_updatefiltered_MR_result:</p>
<img alt="_sources/_static/analysis.png" class="align-center" src="_sources/_static/analysis.png" />
<p>472_updatefiltered_MR_result_indiv:</p>
<img alt="_sources/_static/indiv_analysis.png" class="align-center" src="_sources/_static/indiv_analysis.png" />
<p>We see that there is an estimated effect of 0.367 increase in log odds ratio (a_hat) with an increase in 1 standard deviation of LDL cholesterol levels. The p value is on the order of 10 ^ -4 , indicating a significant effect. This result supports the hypothesis that LDL cholesterol levels causally increase risk for Coronary Heart Disease, matching epidemiological observations.</p>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="../index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">2. Tutorial (Basic Usage)</a><ul>
<li><a class="reference internal" href="#pulling-genetic-instruments-from-gwas-catalog-data">2.1. Pulling Genetic Instruments from GWAS Catalog Data</a><ul>
<li><a class="reference internal" href="#optional-argument">2.1.1. Optional argument:</a></li>
</ul>
</li>
<li><a class="reference internal" href="#selecting-updating-ivs-for-traits-of-interest">2.2. Selecting/Updating IVs for Traits of Interest</a></li>
<li><a class="reference internal" href="#filter-trait-ivfs-for-confounding-and-ld">2.3. Filter Trait IVFs for Confounding and LD</a><ul>
<li><a class="reference internal" href="#nhgri-ignore-and-confounders">2.3.1. nhgri_ignore and confounders</a></li>
<li><a class="reference internal" href="#optional-arguments">2.3.2. Optional arguments:</a></li>
<li><a class="reference internal" href="#units">2.3.3. Units</a></li>
</ul>
</li>
<li><a class="reference internal" href="#calculate-trait-disease-casual-effect-estimate">2.4. Calculate Trait-Disease Casual Effect Estimate</a></li>
</ul>
</li>
</ul>

  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/usage.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li><a href="../index.html">MeRP 0.0.1 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2014, Peter Yin.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.2.2.
    </div>
  </body>
</html>